pyiceberg/io/pyarrow.py

```diff
@@ -1265,6 +1270,7 @@ def _task_to_table(
     case_sensitive: bool,
     name_mapping: Optional[NameMapping] = None,
     use_large_types: bool = True,
+    limit: Optional[int] = None,
 ) -> Optional[pa.Table]:
     batches = list(
```

Comment:
similar to the comment above, i think we can process the `limit` here instead of passing it down to `_task_to_record_batches`